core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -994,6 +1006,139 @@ public void date_format() {
   }
 
+  private FunctionExpression maketime(Expression hour, Expression minute, Expression second) {
+    var repo = new ExpressionConfig().functionRepository();
+    var func = repo.resolve(new FunctionSignature(new FunctionName("maketime"),
+        List.of(DOUBLE, DOUBLE, DOUBLE)));
+    return (FunctionExpression)func.apply(List.of(hour, minute, second));
+  }
+
+  private LocalTime maketime(Double hour, Double minute, Double second) {
+    return maketime(DSL.literal(hour), DSL.literal(minute), DSL.literal(second))
+        .valueOf(null).timeValue();
+  }
+
+  @Test
+  public void maketime() {
+    var r = new Random();
+    assertEquals(maketime(20., 30., 40.), LocalTime.of(20, 30, 40));
+
+    for (var ignored : IntStream.range(0, 25).toArray()) {
+      var hour = r.nextDouble() * 23;
+      var minute = r.nextDouble() * 59;
+      var second = r.nextDouble() * 59;
+      // results could have 1 nanosec diff because of rounding FP
+      var expected = LocalTime.of((int)Math.round(hour), (int)Math.round(minute),
+          // pick fraction second part as nanos
+          (int)Math.floor(second)).withNano((int)((second % 1) * 1E9));
+      var delta = Duration.between(expected, maketime(hour, minute, second)).getNano();
+      assertEquals(0, delta, 1,
+          String.format("hour = %f, minute = %f, second = %f", hour, minute, second));
+    }
+
+    assertEquals(nullValue(), eval(maketime(DSL.literal(-1.), DSL.literal(42.), DSL.literal(42.))),
+        "Negative hour doesn't produce NULL");
+    assertEquals(nullValue(), eval(maketime(DSL.literal(42.), DSL.literal(-1.), DSL.literal(42.))),
+        "Negative minute doesn't produce NULL");
+    assertEquals(nullValue(), eval(maketime(DSL.literal(42.), DSL.literal(42.), DSL.literal(-1.))),
+        "Negative second doesn't produce NULL");
+
+    // The function has 3 arguments, testing NULL and MISSING in all position(s)
+    var combinations = permuteArgumentsWith(3, nullRef, DSL.literal(42.));
+    for (var args : combinations) {
+      assertEquals(nullValue(), eval(maketime(args[0], args[1], args[2])));
+    }
+    combinations = permuteArgumentsWith(3, missingRef, DSL.literal(42.));
+    for (var args : combinations) {
+      assertEquals(missingValue(), eval(maketime(args[0], args[1], args[2])));
+    }
+  }
+
+  private FunctionExpression makedate(Expression year, Expression dayOfYear) {
+    var repo = new ExpressionConfig().functionRepository();
+    var func = repo.resolve(new FunctionSignature(new FunctionName("makedate"),
+        List.of(DOUBLE, DOUBLE)));
+    return (FunctionExpression)func.apply(List.of(year, dayOfYear));
+  }
+
+  private LocalDate makedate(Double year, Double dayOfYear) {
+    return makedate(DSL.literal(year), DSL.literal(dayOfYear)).valueOf(null).dateValue();
+  }
+
+  @Test
+  public void makedate() {
+    var r = new Random();
+    for (var ignored : IntStream.range(0, 1025).toArray()) {
```

Comment:
What cases are tested in this for loop? Why is RNG necessary?

Reply:
RNG helps to find unconsidered cases, it helped me to find few bugs. I dump values which cause a failure, so any other developer would be able to reproduce it.

Reply:
A good discussion on using RNG in unit/integration tests: https://softwareengineering.stackexchange.com/questions/429601/is-the-usage-of-random-values-in-unit-testing-a-good-practice